addon/components/search-result/component.js

```diff
@@ -0,0 +1,170 @@
+import Ember from 'ember';
+import layout from './template';
+import Analytics from '../../mixins/analytics';
+import hostAppName from '../../mixins/host-app-name';
+
+/**
+ * Adapted from Ember-SHARE and Ember Preprints
+ * Used for search results on discover page.
+ *
+ * ```handlebars
+ * {{search-result
+ *      detailRoute=detailRoute
+ *      addFilter='addFilter'
+ *      result=result
+ *      queryParams=queryParams
+ *      filterReplace=filterReplace
+ *      updateFilters=(action 'updateFilters')
+ * }}
+ * ```
+ * @class search-result
+ */
+export default Ember.Component.extend(Analytics, hostAppName, {
+    layout,
+    maxTags: 10,
+    maxSubjects: 10,
+    maxCreators: 10,
+    maxDescription: 300,
+    showBody: false,
+    /**
+     * Array of query params being used in consuming app
+     * @property {Array} queryParams
+     */
+    queryParams: null,
+    /**
+     * Search result from SHARE
+     * @property {Object} result
+     */
+    result: null,
+    providerUrlRegex: {
```

Comment:
With the advent of branding etc, I would encourage moving provider info to a single standard location. Having to maintain the list of providers in many different components could quickly pose a challenge for maintainability.